protocol-units/bridge/move-modules/sources/atomic_bridge_initiator.move

```diff
@@ -0,0 +1,527 @@
+            transfers: vector::empty<BridgeTransfer>(),
+            nonce: 0,
+            bridge_transfer_initiated_events: account::new_event_handle<BridgeTransferInitiatedEvent>(deployer),
+            bridge_transfer_completed_events: account::new_event_handle<BridgeTransferCompletedEvent>(deployer),
+            bridge_transfer_refunded_events: account::new_event_handle<BridgeTransferRefundedEvent>(deployer),
+        });
+        move_to(deployer, BridgeConfig {
+            moveth_minter: signer::address_of(deployer),
+            bridge_module_deployer: signer::address_of(deployer),
+        });
+    }
+
+    public fun initiate_bridge_transfer(
+        initiator: &signer,
+        recipient: vector<u8>, // eth address
+        hash_lock: vector <u8>,
+        time_lock: u64,
+        amount: u64
+    ): vector<u8> acquires BridgeTransferStore, BridgeConfig {
+        let addr = signer::address_of(initiator);
+        let asset = moveth::metadata();
+        let config_address = borrow_global<BridgeConfig>(@atomic_bridge).bridge_module_deployer;
+        let store = borrow_global_mut<BridgeTransferStore>(config_address);
+        //let store = borrow_global_mut<BridgeTransferStore>(addr);
+        assert!(amount > 0, EINSUFFICIENT_AMOUNT);
+        // Transfer amount of moveth from initiator to atomic bridge address
+        let initiator_store = primary_fungible_store::ensure_primary_store_exists(addr, asset);
+
+        // Check balance of initiator account
+        assert!(primary_fungible_store::balance(addr, asset) >= amount, EINSUFFICIENT_BALANCE);
+        let bridge_store = primary_fungible_store::ensure_primary_store_exists(@atomic_bridge, asset);
+        dispatchable_fungible_asset::transfer(initiator, initiator_store, bridge_store, amount);
+        store.nonce = store.nonce + 1;
+
+        // Create a single byte vector by concatenating all components
+        let combined_bytes = vector::empty<u8>();
+        vector::append(&mut combined_bytes, bcs::to_bytes(&addr));
+        vector::append(&mut combined_bytes, recipient);
+        vector::append(&mut combined_bytes, hash_lock);
+        vector::append(&mut combined_bytes, bcs::to_bytes(&store.nonce));
+
+        let bridge_transfer_id = aptos_hash::keccak256(combined_bytes);
+
+        let bridge_transfer = BridgeTransfer {
+            amount: amount,
+            originator: addr,
+            recipient: recipient,
+            hash_lock: hash_lock, 
+            time_lock: timestamp::now_seconds() + time_lock,
+            state: INITIALIZED,
+        };
+
+        vector::push_back(&mut store.transfers, bridge_transfer);
+
+        event::emit_event(&mut store.bridge_transfer_initiated_events, BridgeTransferInitiatedEvent {
+            bridge_transfer_id: bridge_transfer_id,
+            originator: addr,
+            recipient: recipient,
+            amount: amount,
+            hash_lock: hash_lock,
+            time_lock: timestamp::now_seconds() + time_lock,
```

Comment:
we need to update this to use `blockheight` as opposed to timestamp. On solidity side we use blockheight, that's the agreed constant for the bridge's concept of "time" and sync. 

In Initiator .sol:
`timeLock: block.number + timeLock,`

After updating this you'll have to update the test to to kick off blockheights within the test. Let me know if you need a pointer where to find, I saw some code for this in aptos-framework

Reply:
So then `time_lock` will represent a number of blocks rather than a number of seconds, correct?

Reply:
yep, that's right.

Reply:
Okay I'll be afk for about an hour and then will get cracking on this. If you do have a link handy to what you saw in the framework, I'd be interested, thanks.

Once you finish this review pass I'll implement everything and re-submit for review.